legion/engine/core/data/importers/mesh_importers.cpp

```diff
@@ -36,6 +40,141 @@ namespace legion::core::detail
+     * \param bufferView - the tinygltf::BufferView containing information about data size and offset
+     * \param data - std::Vector<T> where the buffer is going to be copied into. The vector will be resized to vector.size()+(tinygltf data size)
+     */
+    template <class T>
+    void handleGltfBuffer(const tinygltf::Buffer& buffer, const tinygltf::BufferView& bufferView, std::vector<T>* data)
+    {
+        /*for (size_t i = 0; i < bufferView.byteLength; i += sizeof(T))
+        {
+            data->push_back(*reinterpret_cast<const T*>(&buffer.data.at(i) + bufferView.byteOffset));
+        }*/
+        size_t size = data->size();
+        data->resize(size + bufferView.byteLength / sizeof(T));
+        memcpy(data->data()+size, &buffer.data.at(0) + bufferView.byteOffset, bufferView.byteLength);
+    }
+
+    /**
+     * @brief Function to handle vertex color of tinygltf
+     * 
+     * \param buffer - tinygltf::Buffer containing the mesh data
+     * \param bufferView - tinygltf::BufferView containing information about the buffer (data size/data offset)
+     * \param accessorType - tinygltf accessorType, Vertex color is expected to come in vec3 or vec4 - will be handled by the function
+     * \param componentType - tinygltf componentType, Vertex color is expected to come in float, unsigned byte or unsigned short - will be handled by the function
+     * \param data - std::vector<color> the destination of the data copy. The vector will be resized to vector.size()+(tinygltf vertex color size)
+     */
+    void handleGltfVertexColor(const tinygltf::Buffer& buffer, const tinygltf::BufferView& bufferView, int accessorType, int componentType, std::vector<math::color>* data)
+    {
+        //colors in glft are in vec3/vec4 float/unsigned byte/unsigned short
+
+        size_t size = data->size();
+        if (accessorType == TINYGLTF_TYPE_VEC3)
+        {
+            // Copy the vertex colors into the vector, keeping in my mind that the vertex color data is only r,g,b
+
+            log::debug("Color accessor type: Vec3");
+            size_t dataIndex = size;
+            if (componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_BYTE)
+            {
+                // Vertex colors in unsigned byte
+                // Currently not supported
+                log::debug("Color component type: u byte");
+                log::warn("Warning: Vert colors for UNSIGNED BYTE not implemented");
+            }
+            else if (componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_SHORT)
+            {
+                // Vertex colors in unsigned short
+                // Currently not supported
+                log::debug("Color component type: u short");
+                log::warn("Warning: Vert colors for UNSIGNED SHORT not implemented");
+            }
+            else if (componentType == TINYGLTF_COMPONENT_TYPE_FLOAT)
+            {
+                // Vertex colors in float
+                log::debug("Color component type: float");
```

Comment:
might wanna remove the debug logs or turn them into trace logs, keep the warnings tho